org.eclipse.dawnsci.analysis.api/src/org/eclipse/dawnsci/analysis/api/persistence/IPersistentFile.java

```diff
@@ -459,5 +459,17 @@
 	 * @throws Exception
 	 */
 	public OriginMetadata getOperationDataOrigin() throws Exception;
+	
+	/**
+	 * Write data and axes in a single call.
+	 * 
+	 * For 1D data set yAxis as null
+	 * 
+	 * @param data
+	 * @param xAxis
+	 * @param yAxis
+	 * @throws Exception
+	 */
+	public void setData(IDataset data, IDataset xAxis, IDataset yAxis) throws Exception;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)